x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -317,7 +327,7 @@ func findPolicy(policies []kibanaPolicy) (*kibanaPolicy, error) {
 func findKey(keys []kibanaAPIKey, policy *kibanaPolicy) (*kibanaAPIKey, error) {
 	tokenName := envWithDefault(defaultTokenName, "FLEET_TOKEN_NAME")
 	for _, key := range keys {
-		name := strings.TrimSpace(strings.Replace(key.Name, fmt.Sprintf(" (%s)", key.ID), "", 1))
+		name := tokenNameStrip.ReplaceAllString(key.Name, "")
```

Comment:
can it be that there will be trailing/leading space after replace as these were removed before

Reply:
nice catch, I added the `strings.TrimSpace` back in